cmd/registry/patterns/resources.go

```diff
@@ -82,6 +87,52 @@ func (s SpecName) ParentName() ResourceName {
 	return VersionName{}
 }
 
+type SpecRevisionName struct {
```

Comment:
Let's not support both Spec and SpecRevision, that will create a lot of confusion.